components/suggest/src/rs.rs

```diff
@@ -93,90 +103,85 @@ impl RemoteSettingsClient {
         })
     }
 
-    fn client_for_record_type(&self, record_type: &str) -> &remote_settings::Client {
-        match record_type {
-            "fakespot-suggestions" => &self.fakespot_client,
+    fn client_for_record(&self, record: &Record) -> &remote_settings::Client {
+        match record.payload {
+            SuggestRecord::Fakespot => &self.fakespot_client,
             _ => &self.quicksuggest_client,
         }
     }
 }
 
 impl Client for RemoteSettingsClient {
-    fn get_records(&self, request: RecordRequest) -> Result<Vec<Record>> {
-        let client = self.client_for_record_type(request.record_type.as_str());
-        let options = request.into();
-        client
-            .get_records_with_options(&options)?
-            .records
-            .into_iter()
-            .map(|record| {
-                let attachment_data = record
-                    .attachment
-                    .as_ref()
-                    .map(|a| client.get_attachment(&a.location))
-                    .transpose()?;
-                Ok(Record::new(record, attachment_data))
-            })
-            .collect()
+    fn get_records(&self, db: &SuggestDb, fetch_fakespot_records: bool) -> Result<Vec<Record>> {
+        let records = get_records_for_collection(&self.quicksuggest_client, db, "quicksuggest")?;
```

Comment:
Non-issue; just out of curiosity (your code is absolutely fine :P). Since iterators are lazy, does the following do the same (assuming `get_records_for_collection` always succeeds)?

```rs
        let records = get_records_for_collection(..., "quicksuggest")?;
        let fsrecords = get_records_for_collection(..., "fakespot-suggest-products")?;

        Ok(if fetch_fakespot_records {
            records.chain(fsrecords).collect()
        } else {
            records.collect()
        })
```

Reply:
Good question!  I *think* this would do some work, since `get_records_for_collection` makes network connections, etc.  It's not like the `async fn` -> `Future` translation which doesn't do anything until the future is first polled.

But even simpler than either of these options is to go with your suggestion to add a collection argument and not special-case fakespot so much.